app/controllers/teams_controller.rb

```diff
@@ -42,11 +42,9 @@ def destroy
 
   def add_user
     authorize team, :update?
-    if includes_user?
-      return redirect_to_edit_team(:error, t('teams.users.add.exists'))
-    end
-    team.users << User.find(team_params[:user_id])
-    redirect_to_edit_team(:success, t('teams.users.add.success'))
+    user = User.find_by_id(team_params[:user_id]) || User.find_by_email(team_params[:email])
+    return add_existent_user(user) if user.present?
+    add_new_invitable_user
```

Comment:
And in this way we would move away code from the controller which is not related to params parsing or params validations